File: lesson_5_1/app/src/main/java/com/kalashnyk/denys/simplemvpapp/data/network/ApiService.java
package com.kalashnyk.denys.simplemvpapp.data.network;

import com.kalashnyk.denys.simplemvpapp.data.database.entity.UserEntity;
import com.kalashnyk.denys.simplemvpapp.data.pojo.response.UserResponse;

import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiService {

    @GET("/api/android/rest/api-v2.php/records/users")
    Single<Response<UserResponse>> getUsers();

    @GET("/api/android/rest/api-v2.php/records/users/{id}")
    Single<UserEntity> getUserById(@Path("id") int id);

}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/repository/server/ApiService.kt
package com.kalashnyk.denys.kotlinbindingsample.repository.server

import com.kalashnyk.denys.kotlinbindingsample.repository.database.entity.UserEntity
import com.kalashnyk.denys.kotlinbindingsample.repository.database.pojo.UserResponse
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {
    @GET("/api/android/rest/api-v2.php/records/users3")
    fun getUsers(): Single<Response<UserResponse>>

    @GET("/api/android/rest/api-v2.php/records/users3/{id}")
    fun getUserById(@Path("id") id: Int): Single<UserEntity>
}
